pmlc/dialect/stripe/tests/translate/double_dot.pbtxt

```diff
@@ -0,0 +1,355 @@
+          attrs { key: "contraction" value {} }
+          attrs { key: "kernel" value {} }
+        }
+        stmts {
+          block {
+            name: "kernel_1(_X2,_X3)"
+            comments: "_X4[x0, x2 : 10, 40] = +(_X2[x0, x1] * _X3[x1, x2])"
+            idxs { name: "x0" range: 10 affine {} }
+            idxs { name: "x1" range: 30 affine {} }
+            idxs { name: "x2" range: 40 affine {} }
+            refs {
+              key: "_X2"
+              value {
+                dir: In
+                from: "_X2"
+                access { terms { key: "x0" value: 1 } }
+                access { terms { key: "x1" value: 1 } }
+                interior_shape {
+                  type: FLOAT32
+                  dims { size: 1 stride: 30 }
+                  dims { size: 1 stride: 1 }
+                }
+                loc {}
+                attrs { key: "contraction" value {} }
+              }
+            }
+            refs {
+              key: "_X3"
+              value {
+                dir: In
+                from: "_X3"
+                access { terms { key: "x1" value: 1 } }
+                access { terms { key: "x2" value: 1 } }
+                interior_shape {
+                  type: FLOAT32
+                  dims { size: 1 stride: 40 }
+                  dims { size: 1 stride: 1 }
+                }
+                loc {}
+                attrs { key: "contraction" value {} }
+              }
+            }
+            refs {
+              key: "_X4"
+              value {
+                dir: Out
+                from: "_X4"
+                access { terms { key: "x0" value: 1 } }
+                access { terms { key: "x2" value: 1 } }
+                interior_shape {
+                  type: FLOAT32
+                  dims { size: 1 stride: 40 }
+                  dims { size: 1 stride: 1 }
+                }
+                agg_op: "add"
+                loc {}
+              }
+            }
+            stmts { load { from: "_X2" into: "$_X2" } }
+            stmts { load { from: "_X3" into: "$_X3" } }
+            stmts {
+              intrinsic {
+                name: "mul"
+                inputs: "$_X2"
+                inputs: "$_X3"
+                outputs: "$_X4"
+                type: FLOAT32
+              }
+            }
+            stmts { store { from: "$_X4" into: "_X4" } }
+            loc {}
+          }
+          attrs { key: "agg_op_add" value {} }
+          attrs { key: "comb_op_mul" value {} }
+          attrs { key: "contraction" value {} }
+          attrs { key: "kernel" value {} }
+        }
+        loc {}
+      }
+      attrs { key: "main" value {} }
+    }
+    loc {}
+  }
+  attrs { key: "program" value {} }
+  attrs { key: "total_macs" value { ival: 18000 } }
+}
+
+# CHECK-LABEL: func @double_dot
+# CHECK-SAME: %arg0: !fp32_2 {stripe.layout = !stripe<"tensor !eltwise.fp32(addr[10:20], addr[20:1])">, stripe.name = "_X0"}
```

Comment:
Hmm, the goal of the test is to cover all the nooks and crannies. This particular test I control the input. I agree in general we should try to make tests resilient. Over time I don't mind changing these tests to be 'looser' but I'm not sure which parts to make loose yet.

Reply:
We can revisit this in a follow-up PR. In general, using regex for variable names is highly recommended and matching only output variables that are captured in a FileCheck variable for the later check of users. If we want to check a lot of small details, I would recommend having multiple simpler tests. In general, we should be careful with CHECK-NEXT. Output IR + CHECK-NEXT all the way would lead to a very flaky test. CHECK-DAG or simply CHECK would be recommended unless we actually want to check that one instruction has to strictly be right after another one. If we want to match a def-use relationship of two ops, capturing the output variable name and their users is usually enough and more flexible. You can also ask for feedback in the LLVM/MLIR mailing list. I'm not sure if these guidelines are written somewhere.